src/sudo/pipeline.rs

```diff
@@ -60,31 +61,12 @@ impl<Policy: PolicyPlugin, Auth: AuthPlugin> Pipeline<Policy, Auth> {
             } => {
                 self.apply_policy_to_context(&mut context, &policy)?;
 
-                let mut auth_status = determine_auth_status(
+                self.auth_and_update_record_file(
```

Comment:
the fact that all the fields of `Authorization::Allowed` are passed into this method to me is an indication that those fields should be their own struct, e.g. `enum Authorization { Allowed(AuthorizationAllowed), /* .. */ }` but that feels out of scope for this PR so I won't refactor that in this PR.

Reply:
Had a similar feeling a while back.